badge_evolution.md

```diff
@@ -0,0 +1,36 @@
+# Badge Feature Evolution Proposal
```

Comment:
There are some good ideas here. Would you mind filing an issue and paste this file into it? It'll get lost if we just commit it into the source tree.